testing/src/scenario/ops_main_mock.py

```diff
@@ -197,11 +189,17 @@ def __init__(
         event: "_Event",
         context: "Context",
         charm_spec: "_CharmSpec[CharmType]",
+        juju_context: Optional[ops.jujucontext._JujuContext] = None,
     ):
         self.state = state
         self.event = event
         self.context = context
         self.charm_spec = charm_spec
+        self.juju_context = (
+            ops.jujucontext._JujuContext.from_dict(os.environ)
+            if juju_context is None
+            else juju_context
+        )
```

Comment:
Style nit: partly for consistency with the `if juju_context is None` statements above, and partly because I think it's clearer (and shorter), I would rather this be reworded like so:
```suggestion
        if juju_context is None:
            juju_context = ops.jujucontext._JujuContext.from_dict(os.environ)
        self.juju_context = juju_context
```